Non real solvability (#3798)
modelica/ModelicaSpecification: chapters/equations.tex

In an equation section, the equations in the body are seen as equations that must be satisfied.
The bodies that are not selected have no effect on that model evaluation.

> The \lstinline!if!-equations in equation sections which do not have exclusively parameter expressions as switching conditions shall satisfy the following:

Reviewer: Let's take the opportunity to get rid of some redundancy, and avoid a potentially confusing reference (equation sections don't have switching conditions):

```suggestion
The \lstinline!if!-equations which do not have exclusively parameter expressions as switching conditions shall satisfy the following:
```

Reviewer: Also, in https://github.com/modelica/ModelicaSpecification/issues/3779#issuecomment-3612632733 we said that this should be tied to evaluability, so I would have expected this:
```suggestion
The \lstinline!if!-equations which do not have exclusively evaluable expressions as switching conditions shall satisfy the following:
```

\begin{itemize}
\item
  Have the same number of equations in each branch (a missing else is counted as zero equations and the number of equations is defined as in \cref{local-equation-size}).
\item
  If the equations are non-Real they shall have component-references (or a list of them) as their left-hand-side.

>   Any indices for such component-references must be evaluable.

Reviewer: I think we call then _subscripts_, compare `\cref{connect-equations-and-connectors}`:

```suggestion
  Any subscripts for such component-references must be evaluable.
```

In an equation section, the equations in the body are seen as equations that must be satisfied.
The bodies that are not selected have no effect on that model evaluation.

The \lstinline!if!-equations which do not have exclusively evaluable expressions as switching conditions shall satisfy the following:
\begin{itemize}
\item
  Have the same number of equations in each branch, where the number of equations is defined as in \cref{local-equation-size}.
  Absence of an \lstinline!else!-branch is treated as having a branch with zero equations.
\item

>   Non-\lstinline!Real! equations in the \lstinline!if!-equation branches shall have component-references (or a list of them) as their left-hand-side.

Reviewer: Hmm… this is one of the examples in `\cref{simple-equality-equations}`:
```
(out1, out2, out3) = function_name(inexpr1, inexpr2);
```
What am I missing?

>   Any subscripts for such component-references must be evaluable.

Reviewer: We also need something for the non-simple equations:

```suggestion
  Any subscripts for such component-references must be evaluable.
  Any \lstinline!for!- and \lstinline!if!-equations in the \lstinline!if!-equation branches shall have evaluable controlling conditions, and contain equations which fullfil these requirements recursively.
```

In an equation section, the equations in the body are seen as equations that must be satisfied.
The bodies that are not selected have no effect on that model evaluation.

The \lstinline!if!-equations which do not have exclusively evaluable expressions as switching conditions shall satisfy the following:
\begin{itemize}
\item
  Have the same number of equations in each branch, where the number of equations is defined as in \cref{local-equation-size}.
  Absence of an \lstinline!else!-branch is treated as having a branch with zero equations.
\item

>   Non-\lstinline!Real! simple equality equations (see \cref{simple-equality-equations}) in the \lstinline!if!-equation branches shall have component-references as their left-hand-side.

Reviewer: Didn't the wrong suggestion get applied here?

```suggestion
  Non-\lstinline!Real! simple equality equations (see \cref{simple-equality-equations}) in the \lstinline!if!-equation branches shall have component-references (or a list of them) as their left-hand-side.
```
